pkg/database/sqlite.go

```diff
@@ -0,0 +1,100 @@
+package database
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/sqlite3"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	_ "github.com/golang-migrate/migrate/v4/source/github"
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/warrant-dev/warrant/pkg/config"
+)
+
+type SQLite struct {
+	SQL
+	Config config.SQLiteConfig
+}
+
+func NewSQLite(config config.SQLiteConfig) *SQLite {
+	return &SQLite{
+		SQL: SQL{
+			DB: nil,
+		},
+		Config: config,
+	}
+}
+
+func (ds SQLite) Type() string {
+	return TypeSQLite
+}
+
+func (ds *SQLite) Connect(ctx context.Context) error {
+	var db *sqlx.DB
+	var err error
+
+	db, err = sqlx.Open("sqlite3", fmt.Sprintf("file:%s?_foreign_keys=on", ds.Config.Database))
+	if err != nil {
+		errors.Wrap(err, "Unable to establish connection to sqlite. Shutting down server.")
+	}
+
+	err = db.PingContext(ctx)
+	if err != nil {
+		errors.Wrap(err, "Unable to ping sqlite. Shutting down server.")
+	}
+
+	// TODO: Configurable max idle conn & max open conn for sqlite?
```

Comment:
You should be able to use the same [`SetMaxOpenConns`](https://pkg.go.dev/database/sql#DB.SetMaxOpenConns) and [`SetMaxIdleConns`](https://pkg.go.dev/database/sql#DB.SetMaxIdleConns) methods for sqlite as well.